File: leetcode35/leetcode35.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
    int low_bound(vector<int>& nums, int target){
        int left = 0, right = nums.size() - 1;
        while (left <= right)
        {
            int mid = left + (right - left) / 2;
            if (nums[mid] >= target){
                right = mid - 1;
            }else{
                left = mid + 1;
            }
        }
        return left;
    }
    public:
        int searchInsert(vector<int>& nums, int target) {
            int ans = low_bound(nums, target);
            return ans;
        }
    };

File: leetcode35/leetcode35.py
class Solution:
    def low_bound(self, nums, target):
        left, right = 0, len(nums) - 1
        while left <= right:
            mid = (left + right) // 2
            if (nums[mid] >= target):
                right = mid - 1
            else:
                left = mid + 1
        return left
    def searchInsert(self, nums: List[int], target: int) -> int:
        return self.low_bound(nums, target)
